File: ICPC/PrimeraFecha2021/pf21/K.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

void solve(){
    string s;
    cin>>s;
    s = '0' + s;
    int size = s.size();
    vector<int> resp;

    if( (s[size - 1]-'0') % 2 == 0)
        resp.push_back(2);
    
    int suma = 0;
    for(char c : s)
        suma = (suma + c-'0')%3;
    if(suma % 3 == 0)
        resp.push_back(3);

    if( (10*(s[size-2]-'0') + (s[size-1]-'0')) % 4 ==0)
        resp.push_back(4);

    if( (s[size-1]-'0') % 5 == 0)
        resp.push_back(5);
    
    if( (suma % 3 ==0) && (s[size - 1] - '0') % 2 == 0)
        resp.push_back(6);

    if(resp.size() == 0){
        cout<<"-1\n";
    }else{
        cout<<resp[0];
        for(int i = 1; i < resp.size(); ++i)
            cout<<" "<<resp[i];
        cout<<"\n";
    }
    
    
}


int main(){
    ios_base::sync_with_stdio(0); cin.tie(0);
    int t; cin>>t;
    for(int i = 0; i < t; ++i)
        solve();
    return 0; 
}

File: TallerProgra/Taller2023/ICPC_1_2021/k.py
def solve():
    b = "0"+input()
    resp =[]

    suma = 0
    for x in b:
        suma =(suma+int(x))%3
    if int(b[-1]) %2 == 0:
        resp.append('2')
    if suma % 3 == 0:
        resp.append('3')
    if int(b[len(b)-2: len(b)]) %4 == 0:
        resp.append('4')
    if b[-1] == '0' or b[-1]=='5':
        resp.append('5')
    if '2' in resp and '3' in resp:
        resp.append('6')
    if len(resp) > 0:
        print(resp[0], end="")
        for i in range(1, len(resp)):
            print(" "+resp[i], end="")
        print("")
    else:
        print("-1")


t = int(input())
for i in range(t):
    solve()